grease/src/Grease/Macaw/Load/Relocation.hs

```diff
@@ -66,57 +64,62 @@ elfRelocationMap _ loadOpts ehi =
+  relAddrType :: EE.RelEntry reloc -> (MM.MemWord w, (reloc, BS.ByteString))
+  relAddrType rel =
+    EE.elfClassInstances elfClass $
       let addr = MM.memWord (offset + fromIntegral (EE.relAddr rel))
-          relType = EE.relType rel in
-      (addr, (relType, resolveRelocSym relType addr (EE.relSym rel)))
+          relType = EE.relType rel
+       in (addr, (relType, resolveRelocSym relType addr (EE.relSym rel)))
 
-    -- The dynamic symbol table, which contains the names of dynamic
-    -- relocations. If the given binary does not have a symbol table, then this
-    -- table will be empty.
-    dynSymtab :: Vec.Vector (EE.SymtabEntry BS.ByteString (EE.ElfWordType w))
-    dynSymtab = case EE.decodeHeaderDynsym ehi of
-                  Just (Right v) -> EE.symtabEntries v
-                  _ -> Vec.empty
+  -- The dynamic symbol table, which contains the names of dynamic
+  -- relocations. If the given binary does not have a symbol table, then this
+  -- table will be empty.
+  dynSymtab :: Vec.Vector (EE.SymtabEntry BS.ByteString (EE.ElfWordType w))
+  dynSymtab = case EE.decodeHeaderDynsym ehi of
+    Just (Right v) -> EE.symtabEntries v
+    _ -> Vec.empty
 
-    -- Look up the name of a dynamic relocation from its dynamic symbol table
-    -- entry. If you reach this point in the code, then we know that the binary
-    -- is dynamically linked (and therefore has a dynamic symbol table), and
-    -- moreover, each dynamic relocation should have a corresponding entry in
-    -- the dynamic symbol table. If it doesn't, this function will raise an
-    -- exception.
-    resolveRelocSym ::
-      Show reloc =>
-      -- | The relocation type (only used for displaying information in an
-      -- exception).
-      reloc ->
-      -- | The relocation's address.
-      MM.MemWord w ->
-      -- | The relocation symbol's index in the dynamic symbol table.
-      Word32 ->
-      BS.ByteString
-    resolveRelocSym reloc addr idx =
-      case dynSymtab Vec.!? fromIntegral idx of
-        Just symb -> EE.steName symb
-        Nothing -> throw $ GreaseException $
-          "Cannot find " <> tshow reloc <>
-          " relocation (at address " <> tshow addr <>
-          ") in the dynamic symbol table"
+  -- Look up the name of a dynamic relocation from its dynamic symbol table
+  -- entry. If you reach this point in the code, then we know that the binary
+  -- is dynamically linked (and therefore has a dynamic symbol table), and
+  -- moreover, each dynamic relocation should have a corresponding entry in
+  -- the dynamic symbol table. If it doesn't, this function will raise an
+  -- exception.
+  resolveRelocSym ::
+    (Show reloc) =>
+    -- \| The relocation type (only used for displaying information in an
+    -- exception).
+    reloc ->
+    -- \| The relocation's address.
+    MM.MemWord w ->
+    -- \| The relocation symbol's index in the dynamic symbol table.
+    Word32 ->
+    BS.ByteString
```

Comment:
Looks like these were unnecessarily escaped too. Maybe let's grep for these escapes to ensure they're all handled before merging?

Reply:
These should be like this `-- ^ The relocation's address.` right?